cpp/src/utils/CMakeLists.txt

```diff
@@ -4,9 +4,13 @@ list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/config.cc)
 list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/file_system.cc)
 list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/model_archive.cc)
 list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/logging.cc)
+list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/metrics/units.cc)
+list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/metrics/dimension.cc)
+list(APPEND TS_UTILS_SOURCE_FILES ${TS_UTILS_SRC_DIR}/metrics/metric.cc)
 add_library(ts_utils SHARED ${TS_UTILS_SOURCE_FILES})
 target_include_directories(ts_utils PUBLIC ${TS_UTILS_SRC_DIR})
-target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS})
+target_include_directories(ts_utils PRIVATE ${Boost_INCLUDE_DIRS})
+target_link_libraries(ts_utils ${FOLLY_LIBRARIES} ${CMAKE_DL_LIBS} ${Boost_LIBRARIES})
```

Comment:
why boost is needed here?

Reply:
Good catch. Boost is not required in this PR, so I'll remove it. I had initially included it to fetch the `hostname` in a platform independent manner while generating the metic log. Will consider including it if required in a subsequent PR when implementing the `STDOUTMetricEmitter` class.

Reply:
On second thought, Boost is required in this PR since I've included implementation for the `Emitter`.